src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -45,18 +47,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         } 
         
         String name = getName(userService.getCurrentUser().getUserId(), datastore);
-        setRegistered(response, name);
+        setRegistered(request, response, name);
     }
 
     /** Determines if a logged in user needs to register, created for testing */
-    public void setRegistered(HttpServletResponse response, String name) throws IOException {
+    public void setRegistered(HttpServletRequest request, HttpServletResponse response, String name) throws IOException {
         LoginStatus loginStatus;
+        // Gets the referring URL of the user so we can return them to the same page when they logout
+        String referrer = request.getHeader("referer");     //"referer" is intentionally misspelled
 
```

Comment:
I couldn't really find where header is being set. Could you explain really briefly how it worls?

Reply:
I'm not setting the header anywhere. getHeader("referer") just returns the address of whatever page the user was at before clicking the logout link.
For further information: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer